core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2496,6 +2545,367 @@ public Type visitGroupingOperation(GroupingOperation node, StackableAstVisitorCo
             }
         }
 
+        @Override
+        public Type visitJsonExists(JsonExists node, StackableAstVisitorContext<Context> context)
+        {
+            List<Type> pathInvocationArgumentTypes = analyzeJsonPathInvocation("JSON_EXISTS", node, node.getJsonPathInvocation(), context);
+
+            // pass remaining information in the node : error behavior
+            ImmutableList.Builder<Type> argumentTypes = ImmutableList.<Type>builder()
+                    .addAll(pathInvocationArgumentTypes)
+                    .add(TINYINT); // enum encoded as integer value
+
+            // resolve function
+            ResolvedFunction function;
+            try {
+                function = plannerContext.getMetadata().resolveFunction(session, QualifiedName.of(JSON_EXISTS_FUNCTION_NAME), fromTypes(argumentTypes.build()));
+            }
+            catch (TrinoException e) {
+                if (e.getLocation().isPresent()) {
```

Comment:
When would this happen?

Reply:
When function is not registered, which shouldn't happen. Should I remove the try-catch?